core/domain/exp_services.py

```diff
@@ -239,19 +239,21 @@ def get_exploration_by_id(exploration_id, strict=True, version=None):
             return None
 
 
-def get_exploration_summary_by_id(exploration_id):
+def get_exploration_summary_by_id(exploration_id, strict=True):
```

Comment:
Have we used `strict` param anywhere except in tests?

Reply:
Yes we do actually in a lot of places. You can grep it. One example is:
https://github.com/oppia/oppia/blob/7ad9d06e434c589f0aaa015252ca65872557b7eb/core/domain/exp_services.py#L209

Reply:
Oh I just realised the link I shared is this function itself. I guess someone changed it

Reply:
@lilithxxx, Sorry for the misunderstanding, I was asking whether we have used `exp_services.get_exploration_summary_by_id(<exp_id>, strict=<bool>)`?

Reply:
So the thing is if we don't use strict, exp_summary_model can never None, hence the lines below would be unreachable(this is because it will throw an error in the first line if there is no exp_summary_model)

Reply:
> hence the lines below would be unreachable

Which line?

Also, by default strict is true, right? and we aren't changing this anywhere in the codebase through `get_exploration_summary_by_id`, so I'm not sure why we need this! (Sorry if I'm missing something!)

Reply:
The lines are:
```
if exp_summary_model:
        exp_summary = get_exploration_summary_from_model(exp_summary_model)
        return exp_summary
```
Exp_summary_model can never be None if we don't use strict

Reply:
If that's the case we should remove `if exp_summary_model:` as we aren't using `strict=False` anywhere to call `get_exploration_summary_by_id`.

OR

Let's change the line 256 with `exploration_id, strict=False)` as we aren't using this method for strict=True anywhere.

(I would prefer the later one! let me know what you think?)

Reply:
I also prefer the later one. Done!